production/ksonnet/loki/multi-zone.libsonnet

```diff
@@ -0,0 +1,201 @@
+{
+  local container = $.core.v1.container,
+  local deployment = $.apps.v1.deployment,
+  local statefulSet = $.apps.v1.statefulSet,
+  local podDisruptionBudget = $.policy.v1beta1.podDisruptionBudget,
+  local volume = $.core.v1.volume,
+  local roleBinding = $.rbac.v1.roleBinding,
+  local role = $.rbac.v1.role,
+  local service = $.core.v1.service,
+  local serviceAccount = $.core.v1.serviceAccount,
+  local servicePort = $.core.v1.servicePort,
+  local policyRule = $.rbac.v1.policyRule,
+  local podAntiAffinity = deployment.mixin.spec.template.spec.affinity.podAntiAffinity,
+
+  _config+: {
+    multi_zone_ingester_enabled: false,
+    multi_zone_ingester_migration_enabled: false,
+    multi_zone_ingester_replicas: 0,
+    multi_zone_ingester_max_unavailable: 25,
+
+    loki+: {
+      ingester+: {
+        lifecycler+: {
+          ring+: (if $._config.multi_zone_ingester_enabled then { zone_awareness_enabled: $._config.multi_zone_ingester_enabled } else {}),
+        } + (if $._config.multi_zone_ingester_enabled then { availability_zone: '${AVAILABILITY_ZONE}' } else {}),
+      },
+    },
+  },
+
+  //
+  // Multi-zone ingesters.
+  //
+
+  ingester_zone_a_args:: {},
+  ingester_zone_b_args:: {},
+  ingester_zone_c_args:: {},
+
+  newIngesterZoneContainer(zone, zone_args)::
+    local zone_name = 'zone-%s' % zone;
+
+    $.ingester_container +
+    container.withArgs($.util.mapToFlags(
+      $.ingester_args + zone_args
+    )) +
+    container.withEnvMixin([{ name: 'AVAILABILITY_ZONE', value: zone_name }]),
```

Comment:
is this from mimir or did you add it yourself? if the latter, do you mind leaving a comment as to what the intended use is for the env var?

Reply:
I am using the same concept we are using in mimir. Since mimir does extra tweaks on top of standard cortex zone awareness, I am using default cortex setting for loki and tempo(https://github.com/grafana/tempo/pull/1936).

Reply:
Sorry if I was unclear, I am asking about the environment variable itself. I don't see a reason why we're adding it.